### PR TITLE
Add telemetry events for interop

### PR DESCRIPTION
**Pull request by idavis**

### CODE REVIEW

**Review by sezna — APPROVED** [review]

File: pip/qsharp/interop/qiskit/backends/backend_base.py [hunk 1]
```diff
@@ -277,7 +277,9 @@ def run_job(
         output["status"] = "COMPLETED"
         output["backend_name"] = self.name
         output["backend_version"] = self.backend_version
-        output["time_taken"] = str(time.time() - start)
+
+        duration = datetime.timedelta(seconds=(monotonic() - start))
```
Comment (billti) [inline pip/qsharp/interop/qiskit/backends/backend_base.py:281]: I assume you verified the resulting string is that same as was previously done on line 280 (or that it's not breaking if not)

File: pip/qsharp/interop/__init__.py [hunk 2]
```diff
@@ -1,2 +1,7 @@
 # Copyright (c) Microsoft Corporation.
 # Licensed under the MIT License.
+
+from .. import telemetry_events
+
+
+telemetry_events.on_qiskit_import()
```
Comment (billti) [inline pip/qsharp/interop/__init__.py:7]: Do we care about them just importing the 'interop' module?

File: pip/qsharp/interop/qiskit/jobs/qsjob.py [hunk 3]
```diff
@@ -15,7 +16,7 @@
 from qiskit.providers import JobV1, JobStatus, JobError
 
 from ..execution import DetaultExecutor
-
+from ... import telemetry_events
```
Comment (billti) [inline pip/qsharp/interop/qiskit/jobs/qsjob.py:19]: Is 3 dots enough here? Seems like that would get you to the ./interop dir.

File: pip/qsharp/interop/qiskit/jobs/qsjobset.py [hunk 4]
```diff
@@ -17,6 +18,7 @@
 
 from .qsjob import QsSimJob, RunInputCallable
 from ..execution import DetaultExecutor
+from ... import telemetry_events
```
Comment (billti) [inline pip/qsharp/interop/qiskit/jobs/qsjobset.py:21]: Weird if it's working. Maybe I just don't understand the syntax well enough. Seems like one dot is current (jobs) dir, two would be 'qiskit' dir, and three would only get you to 'interop' dir. Not sure why it doesn't need four to get to the root from here.

File: pip/qsharp/telemetry_events.py [hunk 5]
```diff
@@ -85,3 +85,42 @@ def on_estimate_end(durationMs: float, qubits: Union[str, int]) -> None:
...
+
+def on_qiskit_import() -> None:
+    log_telemetry("qiskit.import", 1)
+
+
+def on_qiskit_run(shots: int, num_circuits: int) -> None:
+    log_telemetry(
+        "qiskit.run",
+        1,
+        properties={"shots": get_shots_bucket(shots), "num_circuits": num_circuits},
```
Comment (billti) [inline pip/qsharp/telemetry_events.py:101]: What is the range for 'num_circuits'? Properties on metrics should be a limited set of value (which is why we bucket shots and qubits). Could this end up being dozens (or more) of distinct values?

File: pip/qsharp/interop/qiskit/jobs/qsjobset.py (hunk 4 above)
Comment (idavis) [reply pip/qsharp/interop/qiskit/jobs/qsjobset.py:21]: Two or more leading dots indicate a relative import to the parent(s) of the current package, one level per dot after the first. The parent of the current package is `qiskit`.  Two levels up (due to three dots) gets us to the root package.

File: pip/qsharp/interop/qiskit/backends/backend_base.py (hunk 1 above)
Comment (idavis) [reply pip/qsharp/interop/qiskit/backends/backend_base.py]: It was on my list to double check. You were right, the old value was float in secs. Could have sworn `time()` returned a struct for the `-` to give formatting.

File: pip/qsharp/interop/qiskit/jobs/qsjobset.py (hunk 4 above)
Comment (billti) [reply pip/qsharp/interop/qiskit/jobs/qsjobset.py:21]: Pretty sure 3 dots isn't getting you to the root level. It's getting you to qsharp/interop. However, in qsharp/interop/__init__.py you have `from .. import telemetry_events` so it's finding that re-export.

Comment (idavis) [reply pip/qsharp/interop/qiskit/jobs/qsjobset.py:21]: You are correct

**Review by billti — APPROVED** [review]

**Pull request merged**